app/src/main/java/org/mozilla/gecko/search/SearchWidgetProvider.kt

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.gecko.search
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import mozilla.components.feature.search.widget.AppSearchWidgetProvider
+import mozilla.components.feature.search.widget.BaseVoiceSearchActivity
+import mozilla.components.feature.search.widget.SearchWidgetConfig
+import mozilla.components.support.utils.PendingIntentUtils
+import org.mozilla.focus.R
+import org.mozilla.focus.activity.IntentReceiverActivity
+import org.mozilla.focus.ext.components
+import org.mozilla.focus.searchwidget.VoiceSearchActivity
+
+class SearchWidgetProvider : AppSearchWidgetProvider() {
+
+    // Implementation note:
+    // This class name (SearchWidgetProvider) and package name (org.mozilla.gecko.search) should
+    // not be changed because otherwise this widget will disappear from the home screen of the user.
+    // The existing name replicates the name and package we used in Fennec.
+    override fun onEnabled(context: Context) {
+        context.components.settings.addSearchWidgetInstalled(1)
+    }
+
+    override fun onDeleted(context: Context, appWidgetIds: IntArray) {
+        context.components.settings.addSearchWidgetInstalled(-appWidgetIds.size)
+    }
+
+    override val config: SearchWidgetConfig =
+        SearchWidgetConfig(
+            searchWidgetIconResource = R.drawable.ic_splash_screen,
+            searchWidgetMicrophoneResource = R.drawable.mozac_ic_microphone,
+            appName = R.string.app_name
+        )
+
+    override fun createTextSearchIntent(context: Context): PendingIntent {
+        return Intent(context, IntentReceiverActivity::class.java)
+            .let { intent ->
```

Comment:
nit: The `let` usage and formatting caught my attention. The construction of the PendingIntent seems a bit confusing.
Can we split this in two steps: first constructing the Intent with maybe using `apply` and finally building and returning the PendingIntent?